internal/controller/garbage_collector_runner.go

```diff
@@ -0,0 +1,135 @@
+}
+
+func NewGarbageCollectorRunner(
+	client client.Client,
+	interval time.Duration,
+	managedCephClusterID string,
+) *GarbageCollectorRunner {
+	return &GarbageCollectorRunner{
+		client:               client,
+		interval:             interval,
+		managedCephClusterID: managedCephClusterID,
+	}
+}
+
+func (r *GarbageCollectorRunner) Start(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	for {
+		ctxSleep, cancelSleep := context.WithTimeout(ctx, r.interval)
+		<-ctxSleep.Done()
+		cancelSleep()
+		if ctx.Err() != nil {
+			break
+		}
+
+		logger.Info("garbage collection started")
+
+		if err := r.deleteOrphanedPVs(ctx); err != nil {
+			logger.Error(err, "failed to delete orphaned PVs", "error", err)
+		}
+
+		logger.Info("garbage collection finished")
+	}
+
+	return nil
+}
+
+func (r *GarbageCollectorRunner) deleteOrphanedPVs(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	requirement, err := labels.NewRequirement(labelRestoringPVKey, selection.Exists, []string{})
+	if err != nil {
+		return fmt.Errorf("failed to create a new labels requirement: %w", err)
+	}
+	selector := labels.ValidatedSetSelector{}.Add(*requirement)
+
+	var pvList corev1.PersistentVolumeList
+	if err := r.client.List(ctx, &pvList, &client.ListOptions{
+		LabelSelector: selector,
+	}); err != nil {
+		return fmt.Errorf("failed to list PVs: %w", err)
+	}
+
+	for _, pv := range pvList.Items {
+		shouldDelete, err := r.isMantleRestoreAlreadyDeleted(ctx, &pv)
+		if err != nil {
+			return fmt.Errorf("failed to check if a PV should be deleted: %w", err)
+		}
+		if !shouldDelete {
+			continue
+		}
+		if err := r.client.Delete(ctx, &pv, &client.DeleteOptions{
+			Preconditions: &metav1.Preconditions{UID: &pv.ObjectMeta.UID, ResourceVersion: &pv.ObjectMeta.ResourceVersion},
+		}); err != nil {
+			return fmt.Errorf("failed to delete PV: %w", err)
+		}
+		logger.Info("an orphaned PV is removed", "name", pv.GetName())
+	}
+
+	return nil
+}
+
+func (r *GarbageCollectorRunner) isMantleRestoreAlreadyDeleted(ctx context.Context, pv *corev1.PersistentVolume) (bool, error) {
+	restoreUID, ok := pv.GetAnnotations()[PVAnnotationRestoredBy]
+	if !ok {
+		return false, fmt.Errorf("failed to find annotation: %s: %s", PVAnnotationRestoredBy, pv.GetName())
+	}
+	restoreName, ok := pv.GetAnnotations()[PVAnnotationRestoredByName]
+	if !ok {
+		return false, fmt.Errorf("failed to find annotation: %s: %s", PVAnnotationRestoredByName, pv.GetName())
+	}
+	restoreNamespace, ok := pv.GetAnnotations()[PVAnnotationRestoredByNamespace]
+	if !ok {
+		return false, fmt.Errorf("failed to find annotation: %s: %s", PVAnnotationRestoredByNamespace, pv.GetName())
+	}
+
+	clusterID, ok := pv.Spec.CSI.VolumeAttributes["clusterID"]
+	if !ok {
+		return false, fmt.Errorf("failed to find cluster ID: %s", pv.GetName())
+	}
+	if r.managedCephClusterID != clusterID {
+		return false, fmt.Errorf("PV is not managed by target Ceph cluster: %s: %s", clusterID, r.managedCephClusterID)
```

Comment:
This is not an error case.

```suggestion
		return false, nil
```

Reply:
Fixed in https://github.com/cybozu-go/mantle/pull/71/commits/1d60ebc98f03f7ac21ccc914788669e3a9cb1005.